static/js/components/email/hoc.js

```diff
@@ -0,0 +1,120 @@
+      }
+
+      getActiveEmailState(): EmailState {
+        const { email } = this.props;
+        return email[email.currentlyActive];
+      }
+
+      openEmailComposer = R.curry((emailType: string, emailOpenParams: any): void => {
+        const { dispatch } = this.props;
+        dispatch(
+          startEmailEdit({
+            type: emailType,
+            ...emailConfigs[emailType].emailOpenAction(emailOpenParams)
+          })
+        );
+        dispatch(showDialog(EMAIL_COMPOSITION_DIALOG));
+      });
+
+      updateEmailFieldEdit = R.curry((fieldName, e): void => {
+        const { dispatch, email: { currentlyActive } } = this.props;
+        let activeEmail = this.getActiveEmailState();
+        let inputsClone = R.clone(activeEmail.inputs);
+        inputsClone[fieldName] = e.target.value;
+        dispatch(updateEmailEdit({type: currentlyActive, inputs: inputsClone}));
+        if (!R.isEmpty(activeEmail.validationErrors)) {
+          let cloneErrors = emailValidation(inputsClone);
+          dispatch(updateEmailValidation({type: currentlyActive, errors: cloneErrors}));
+        }
+      });
+
+      closeAndClearEmailComposer = (): void => {
+        const { dispatch, email: { currentlyActive } } = this.props;
+        dispatch(clearEmailEdit(currentlyActive));
+        dispatch(hideDialog(EMAIL_COMPOSITION_DIALOG));
+      };
+
+      closeEmailComposerAndSend = (): void => {
+        const { dispatch, email: { currentlyActive } } = this.props;
+        let activeEmail = this.getActiveEmailState();
+        let errors = emailValidation(activeEmail.inputs);
+        dispatch(updateEmailValidation({type: currentlyActive, errors: errors}));
+        if (R.isEmpty(errors)) {
+          dispatch(
+            emailConfigs[currentlyActive].emailSendAction(activeEmail)
+          ).then(() => {
+            this.closeAndClearEmailComposer();
+          });
+        }
+      };
+
+      renderCompositionDialog(): ?React$Element<*> {
+        let { ui: { dialogVisibility } } = this.props;
+
+        let activeEmailType = this.getActiveEmailType();
+
+        if (!activeEmailType) return null;
+
+        return (
+          <EmailCompositionDialog
+            updateEmailFieldEdit={this.updateEmailFieldEdit}
+            closeAndClearEmailComposer={this.closeAndClearEmailComposer}
+            closeEmailComposerAndSend={this.closeEmailComposerAndSend}
+            dialogVisibility={dialogVisibility[EMAIL_COMPOSITION_DIALOG]}
+            activeEmail={this.getActiveEmailState()}
+            title={emailConfigs[activeEmailType].title}
+            subheadingRenderer={emailConfigs[activeEmailType].renderSubheading}
+          />
+        );
+      }
+
+      render() {
+        let emailOpenActions = {};
+        for (let emailType of Object.keys(emailConfigs)) {
+          emailOpenActions[emailType] = this.openEmailComposer(emailType);
+        }
```

Comment:
it seems like we could maybe avoid this step and just pass down `this.openEmailComposer`, trusting the the call-site to specify the right `emailType`